tests/common.go

```diff
@@ -7728,3 +7781,283 @@ func GetPoolCapacityUsed(poolUUID string) (float64, error) {
+			defer ginkgo.GinkgoRecover()
+			defer wg.Done()
+
+			poolID, err := GetPoolIDFromPoolUUID(poolUuid)
+			if err != nil {
+				errCh <- err
+				return
+			}
+
+			inspectVolume, err := Inst().V.InspectVolume(v.ID)
+			if err != nil {
+				errCh <- err
+				return
+			}
+			log.InfoD("Getting md5sum for volume %s on pool %s in node %s", inspectVolume.Id, poolUuid, nodeDetail.Name)
+			//To-Do update the command if set up is dmthin
+			cmd := fmt.Sprintf("/opt/pwx/bin/runc exec -t portworx md5sum /var/.px/%d/%s/pxdev", poolID, inspectVolume.Id)
+			isDmthinSetup, err := IsDMthin()
+			if err != nil {
+				errCh <- err
+				return
+			}
+
+			if isDmthinSetup {
+				cmd = fmt.Sprintf("md5sum /dev/pwx%d/%s", poolID, inspectVolume.Id)
+			}
+			output, err := Inst().N.RunCommand(*nodeDetail, cmd, node.ConnectionOpts{
+				Timeout:         defaultTimeout,
+				TimeBeforeRetry: defaultRetryInterval,
+				Sudo:            true,
+			})
+			if err != nil {
+				errCh <- err
+				return
+			}
+			vChecksum := strings.Split(strings.TrimSpace(output), " ")[0]
+			poolChecksumMap[poolUuid] = vChecksum
+		}
+
+		for _, v := range appVolumes {
+			replicaSets, err := Inst().V.GetReplicaSets(v)
+			if err != nil {
+				return err
+			}
+
+			wGroup := new(sync.WaitGroup)
+			errCh := make(chan error)
+
+			for _, rs := range replicaSets {
+				poolUuids := rs.PoolUuids
+				for _, poolUuid := range poolUuids {
+					nodeDetail, err := GetNodeWithGivenPoolID(poolUuid)
+					if err != nil {
+						return err
+					}
+					wGroup.Add(1)
+					go calChecksum(wGroup, v, nodeDetail, poolUuid, errCh)
+				}
+			}
+			wGroup.Wait()
+			close(errCh)
+
+			var errList []error
+
+			for err := range errCh {
+				errList = append(errList, err)
+			}
+
+			if len(errList) != 0 {
+				// Combine all errors into one and return
+				return errors.New(fmt.Sprintf("multiple errors occurred while checking md5sum of volume [%s]", v.ID)).(error)
+			}
+
+			var checksum string
+			var primaryPool string
+			for p, c := range poolChecksumMap {
+				if checksum == "" {
+					primaryPool = p
+					checksum = c
+				} else {
+					if c != checksum {
+						return fmt.Errorf("md5sum of volume [%s] having [%s] on pool [%s] is not matching with checksum [%s] on pool [%s]", v.ID, checksum, primaryPool, c, p)
+					}
+					log.InfoD("md5sum of volume [%s] having [%s] on pool [%s] is matching with checksum [%s] on pool [%s]", v.ID, checksum, primaryPool, c, p)
+				}
+
+			}
+			//clearing the pool after the volume validation
+			for k := range poolChecksumMap {
+				delete(poolChecksumMap, k)
+			}
+
+		}
+
+		//reverting application scale
+		applicationScaleUpMap := make(map[string]int32, len(ctx.App.SpecList))
+
+		for name, scale := range applicationScaleMap {
+			log.InfoD(fmt.Sprintf("scale up app %s to %d", ctx.App.Key, scale))
+			applicationScaleUpMap[name] = scale
+		}
+		err = Inst().S.ScaleApplication(ctx, applicationScaleUpMap)
+		if err != nil {
+			return err
+		}
```

Comment:
scaling all apps at a  time can lead to push lots of data at a time, which can lead to high latency on the pool . can we scaleup corresponding app just after checksum verification so that we donot lots of IO at a time.

Reply:
yes it scales only the app which we are doing checksum validation